integrationTests/testProcessorNode.go

```diff
@@ -0,0 +1,2497 @@
+}
+
+// GetBlockBody returns the body for provided header parameter
+func (tpn *TestProcessorNode) GetBlockBody(header data.HeaderHandler) (*dataBlock.Body, error) {
+	invalidCachers := tpn.DataPool == nil || tpn.DataPool.MiniBlocks() == nil
+	if invalidCachers {
+		return nil, errors.New("invalid data pool")
+	}
+
+	body := &dataBlock.Body{}
+	for _, miniBlockHeader := range header.GetMiniBlockHeaderHandlers() {
+		miniBlockHash := miniBlockHeader.GetHash()
+
+		mbObject, ok := tpn.DataPool.MiniBlocks().Get(miniBlockHash)
+		if !ok {
+			return nil, errors.New(fmt.Sprintf("no miniblock found for hash %s", hex.EncodeToString(miniBlockHash)))
+		}
+
+		mb, ok := mbObject.(*dataBlock.MiniBlock)
+		if !ok {
+			return nil, errors.New(fmt.Sprintf("not a *dataBlock.MiniBlock stored in miniblocks found for hash %s", hex.EncodeToString(miniBlockHash)))
+		}
+
+		body.MiniBlocks = append(body.MiniBlocks, mb)
+	}
+
+	return body, nil
+}
+
+// SyncNode tries to process and commit a block already stored in data pool with provided nonce
+func (tpn *TestProcessorNode) SyncNode(nonce uint64) error {
+	return tpn.syncShardNode(nonce)
+}
+
+func (tpn *TestProcessorNode) syncShardNode(nonce uint64) error {
+	header, err := tpn.GetShardHeader(nonce)
+	if err != nil {
+		return err
+	}
+
+	body, err := tpn.GetBlockBody(header)
+	if err != nil {
+		return err
+	}
+
+	err = tpn.BlockProcessor.ProcessBlock(
+		header,
+		body,
+		func() time.Duration {
+			return time.Second * 5
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	err = tpn.BlockProcessor.CommitBlock(header, body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SetAccountNonce sets the account nonce with journal
+func (tpn *TestProcessorNode) SetAccountNonce(nonce uint64) error {
+	nodeAccount, _ := tpn.AccntState.LoadAccount(tpn.OwnAccount.Address)
+	nodeAccount.(state.UserAccountHandler).IncreaseNonce(nonce)
+
+	err := tpn.AccntState.SaveAccount(nodeAccount)
+	if err != nil {
+		return err
+	}
+
+	_, err = tpn.AccntState.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// MiniBlocksPresent checks if the all the miniblocks are present in the pool
+func (tpn *TestProcessorNode) MiniBlocksPresent(hashes [][]byte) bool {
+	mbCacher := tpn.DataPool.MiniBlocks()
+	for i := 0; i < len(hashes); i++ {
+		ok := mbCacher.Has(hashes[i])
+		if !ok {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (tpn *TestProcessorNode) initRoundHandler() {
+	tpn.RoundHandler = &mock.RoundHandlerMock{TimeDurationField: 5 * time.Second}
+}
+
+func (tpn *TestProcessorNode) initRequestedItemsHandler() {
+	tpn.RequestedItemsHandler = timecache.NewTimeCache(roundDuration)
+}
+
+func (tpn *TestProcessorNode) initBlockTracker() {
+	argBaseTracker := track.ArgBaseTracker{
+		Hasher:           TestHasher,
+		HeaderValidator:  tpn.HeaderValidator,
+		Marshalizer:      TestMarshalizer,
+		RequestHandler:   tpn.RequestHandler,
+		RoundHandler:     tpn.RoundHandler,
+		ShardCoordinator: tpn.ShardCoordinator,
```

Comment:
we have to trick this component to "think" it's part of the metachain

Reply:
we will have NewShardBlockTrackerForSideChain :P or whatever new name.